quinn-udp/src/unix.rs

```diff
@@ -24,6 +24,38 @@ use super::{
     IO_ERROR_LOG_INTERVAL,
 };
 
+// Adapted from https://github.com/apple-oss-distributions/xnu/blob/main/bsd/sys/socket.h
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+#[repr(C)]
+#[allow(non_camel_case_types)]
+pub(crate) struct msghdr_x {
+    pub msg_name: *mut std::ffi::c_void,
```

Comment:
Looks like so far we've imported symbols from `std::ffi` directly, keeping `libc` symbols as prefixed. I'd be in favor of sticking with that style.